jepostule/pipeline/tests/test_events.py

```diff
@@ -0,0 +1,136 @@
+            employer_email='boss@company.com',
+            candidate_email='candidate@pe.fr',
+            client_platform=self.client_platform,
+        )
+        event1 = job_application.events.create(
+            name=models.JobApplicationEvent.SENT_TO_EMPLOYER
+        )
+        event2 = job_application.events.create(
+            name=models.JobApplicationEvent.CONFIRMED_TO_CANDIDATE
+        )
+        event3 = job_application.events.create(
+            name=models.JobApplicationEvent.ANSWERED
+        )
+        self.assertEqual('boss@company.com', event1.to_email)
+        self.assertEqual('candidate@pe.fr', event2.to_email)
+        self.assertEqual('candidate@pe.fr', event3.to_email)
+
+    def test_large_message_id(self):
+        job_application = models.JobApplication.objects.create(
+            employer_email='boss@company.com',
+            candidate_email='candidate@pe.fr',
+            client_platform=self.client_platform,
+        )
+        event = job_application.events.create(
+            name=models.JobApplicationEvent.ANSWERED,
+        )
+        models.Email.objects.create(
+            event=event,
+            message_id=576460752697195324,
+        )
+
+    def test_open_event(self):
+        job_application = models.JobApplication.objects.create(
+            employer_email='boss@company.com',
+            candidate_email='candidate@pe.fr',
+            client_platform=self.client_platform,
+        )
+        event = job_application.events.create(
+            name=models.JobApplicationEvent.SENT_TO_EMPLOYER
+        )
+        models.Email.objects.create(event=event, status=models.Email.SENT, message_id=123)
+        events.receive({
+            'event': 'open', 'time': time(),
+            'MessageID': 123,
+            'email': 'drunk@propaganda.cn',
+        })
+        events.process.consume()
+        self.assertEqual(models.Email.OPENED, models.Email.objects.get().status)
+        self.assertFalse(blacklist.is_blacklisted("drunk@propaganda.cn"))
+
+    def test_spam_event(self):
+        event = {
+            'event': 'spam', 'time': time(),
+            'MessageID': 58265320973912910,
+            'email': 'drunk@propaganda.cn',
+        }
+        events.receive(event)
+        events.process.consume()
+        self.assertTrue(blacklist.is_blacklisted("drunk@propaganda.cn"))
+
+    @override_settings(BLACKLIST_DURATION_SECONDS=1)
+    def test_spam_event_in_the_past(self):
+        event = {
+            'event': 'spam', 'time': time() - 2,
+            'MessageID': 58265320973912910,
+            'email': 'drunk@propaganda.cn',
+        }
+        events.receive(event)
+        events.process.consume()
+        self.assertFalse(blacklist.is_blacklisted("drunk@propaganda.cn"))
+
+    def test_spam_event_incorrect_timestamp(self):
+        event = {
+            'event': 'spam', 'time': 'wtf',
+            'MessageID': 58265320973912910,
```

Comment:
DNRY ? this value appears many times
or why not just 12345 instead of this very specific value?

Reply:
Message IDs have large values that may not be stored as int, but long. So it makes sense to have long values in the tests. That particular test does not test that, though, so I guess we could use something else.

My opinion is that DNRY does not apply 100% in tests ; otherwise, the code to run test would become quite complex and would have to be tested, too. So I have no issue copy-pasting test code.

Reply:
all right